azurecontainerapps.ts

```diff
@@ -232,9 +232,36 @@ export class azurecontainerapps {
     private static async getLocation(): Promise<string> {
         // Set deployment location, if provided
         let location: string = this.toolHelper.getInput('location', false);
+        let resourceGroup: string = this.toolHelper.getInput('resourceGroup', false);
+        let containerAppEnvironment: string = this.toolHelper.getInput('containerAppEnvironment', false);
 
-        // If no location was provided, use the default location for the Container App service
+        // If no location was provided, attempt to discover the location of the existing Container App Environment linked to the Container App
+        // or Container App Environment provided in the resource group or use the default location.
         if (this.util.isNullOrEmpty(location)) {
+            if (!this.util.isNullOrEmpty(resourceGroup)) {
+                // Check if Container App exists in the resource group provided and get the location from the Container App Environment linked to it
+                let doesContainerAppExist = await this.appHelper.doesContainerAppExist(this.containerAppName, resourceGroup);
+                if (doesContainerAppExist) {
+                    // Get the name of the Container App Environment linked to the Container App
+                    var environmentName = await this.appHelper.getExistingContainerAppEnvironmentName(this.containerAppName, resourceGroup);
+                    // Check if environment exists in the resource group provided and get the location
+                    var doesContainerAppEnvironmentExistInResourceGroup = !this.util.isNullOrEmpty(environmentName) ? await this.appHelper.doesContainerAppEnvironmentExist(environmentName, resourceGroup) : null;
+                    if (doesContainerAppEnvironmentExistInResourceGroup) {
+                        // Get the location of the Container App Environment linked to the Container App
+                        location = await this.appHelper.getExistingContainerAppEnvironmentLocation(environmentName, resourceGroup);
+                        return location;
+                    }
+                }
+
+                // Check if Container App Environment is provided and exits in the resource group provided and get the location
+                let doesContainerAppEnvironmentExist = !this.util.isNullOrEmpty(containerAppEnvironment) ? await this.appHelper.doesContainerAppEnvironmentExist(containerAppEnvironment, resourceGroup) : null;
```

Comment:
Could you please confirm that at this point we know that containerAppEnvironment is set? I can't find the condition validating it.

Reply:
This line of code `let containerAppEnvironment: string = this.toolHelper.getInput('containerAppEnvironment', false);` is executed before this code on line 257. So, we are checking if the `containerAppEnvironment` argument was provided by the customer and exists in the resource group. 

If it does not, we use the default location.